nixos/modules/services/misc/transfer-sh.nix

```diff
@@ -0,0 +1,434 @@
+        type = types.str;
+      };
+
+      group = mkOption {
+        description = "group to run as";
+        default = "transfersh";
+        type = types.str;
+      };
+
+      provider = mkOption {
+        description = "which storage provider to use (s3, storj, gdrive or local)";
+        default = "local";
+        type = types.enum [ "s3" "storj" "gdrive" "local" ];
+      };
+
+      address = mkOption {
+        description = "address to listen on";
+        default = "127.0.0.1";
+        type = types.string;
+      };
+
+      LISTENER = mkOption {
+        description = "port to use for http";
+        default = 6080;
+        type = types.nullOr types.int;
+      };
+
+      PROFILE_LISTENER = mkOption {
+        description = "port to use for profiler";
+        default = 6060;
+        type = types.nullOr types.int;
+      };
+
+      FORCE_HTTPS = mkOption {
+        description = "redirect to https";
+        default = "false";
+        type = types.nullOr types.str;
+      };
+
+      TLS_LISTENER = mkOption {
+        description = "port to use for https";
+        default = null;
+        type = types.nullOr types.int;
+      };
+
+      TLS_LISTENER_ONLY = mkOption {
+        description = "flag to enable tls listener only";
+        default = "false";
+        type = types.nullOr types.str;
+      };
+
+      TLS_CERT_FILE = mkOption {
+        description = "path to tls certificate";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      TLS_PRIVATE_KEY = mkOption {
+        description = "path to tls private key";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      HTTP_AUTH_USER = mkOption {
+        description = "user for basic http auth on upload";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      HTTP_AUTH_PASS = mkOption {
+        description = "pass for basic http auth on upload";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      HTTP_AUTH_HTPASSWD = mkOption {
+        description = "htpasswd file path for basic http auth on upload";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      HTTP_AUTH_IP_WHITELIST = mkOption {
+        description = "comma separated list of ips allowed to upload without being challenged an http auth";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      IP_WHITELIST = mkOption {
+        description = "comma separated list of ips allowed to connect to the service";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      IP_BLACKLIST = mkOption {
+        description = "comma separated list of ips not allowed to connect to the service";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      TEMP_PATH = mkOption {
+        description = "path to temp folder";
+        default = null;
+        type = types.nullOr types.str;
+      };
```

Comment:
This should just be hardcoded in the systemd unit

Reply:
Not sure how to do this. Isn't this user-configurable, how can I set it in the unit?